examples/preprocessing/plot_eeg_csd.py

```diff
@@ -0,0 +1,107 @@
+"""
+=====================================================
+Transform EEG data using current source density (CSD)
+=====================================================
+
+This script shows an example of how to use CSD [1]_ [2]_ [3]_.
+CSD takes the spatial Laplacian of the sensor signal (derivative in both
+x and y). It does what a planar gradiometer does in MEG. Computing these
+spatial derivatives reduces point spread. CSD transformed data have a sharper
+or more distinct topography, reducing the negative impact of volume conduction.
+"""
+# Authors: Alex Rockhill <aprockhill206@gmail.com>
+#
+# License: BSD (3-clause)
+
+# sphinx_gallery_thumbnail_number = 6
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+import mne
+from mne.datasets import sample
+
+print(__doc__)
+
+data_path = sample.data_path()
+
+###############################################################################
+# Load sample subject data
+raw = mne.io.read_raw_fif(data_path + '/MEG/sample/sample_audvis_raw.fif',
+                          preload=True)
+events = mne.find_events(raw)
+raw = raw.pick_types(meg=False, eeg=True, eog=True, ecg=True, stim=False,
+                     exclude=raw.info['bads'])
+raw.set_eeg_reference(projection=True).apply_proj()
+
+###############################################################################
+# Plot the raw data and CSD-transformed raw data:
+
+raw_csd = mne.preprocessing.compute_current_source_density(raw)
+raw.plot()
+raw_csd.plot()
+
+###############################################################################
+# Also look at the power spectral densities:
+
+raw.plot_psd()
+raw_csd.plot_psd()
+
+###############################################################################
+# CSD can also be computed on Evoked (averaged) data.
+# Here we epoch and average the data so we can demonstrate that.
+
+event_id = {'auditory/left': 1, 'auditory/right': 2, 'visual/left': 3,
+            'visual/right': 4, 'smiley': 5, 'button': 32}
+epochs = mne.Epochs(raw, events, event_id=event_id, tmin=-0.2, tmax=.5,
+                    preload=True)
+evo = epochs['auditory'].average()
+
+###############################################################################
+# First let's look at how CSD affects scalp topography:
+
+fig, axes = plt.subplots(1, 6)
+times = np.array([-0.1, 0., 0.05, 0.1, 0.15])
+evo.plot_topomap(times=times, cmap='Spectral_r', axes=axes[:5],
+                 outlines='skirt', contours=4, time_unit='s',
+                 colorbar=True, show=False, title='Average Reference')
+fig, axes = plt.subplots(1, 6)
+csd_evo = mne.preprocessing.compute_current_source_density(evo)
```

Comment:
```suggestion
csd_evoked = mne.preprocessing.compute_current_source_density(evoked)
```

this @alexrockhill this is not resolved. Also I still don't get why you have 2 examples one of which (this one) seems to be more comprehensive. plot_csd.py does not seem to bring anything more.